### DISCUSSION TITLE
Why renaming methods?

### DISCUSSION THREAD

**Discussion by dolfinus** [pydantic/pydantic#4961] [Pydantic V2]
Why methods like .dict(), .json(), .json_schema() were renamed in V2 to .model_dump(), .model_dump_json(), .model_json_schema()?

This breaks backward compatibility a lot, and can be easily avoided. Also new names are very ugly, especially model_json_schema.

**Comment by samuelcolvin** [comment]
https://docs.pydantic.dev/blog/pydantic-v2/#model-namespace-cleanup

I disagree that they're ugly.

We should add back the old methods as simple proxies with depreciation warnings though.

**Comment by VianneyMI** [comment]
Yes please leave the old names for now, for backward compatibility. @samuelcolvin 
I agree with @dolfinus , the new names if not ugly are verbose.
I followed the thread you mentionned @samuelcolvin  and it seemed that most people didn't see a problem. So if it is not broken why fix it ?

**Reply by dmontagu** [reply]
@VianneyMI I think the biggest issue is field shadowing, as noted [here](https://github.com/pydantic/pydantic/issues/1001#issuecomment-1268279072).

Renaming the methods into a "namespace" (i.e., requiring the `model_` prefix) retains the advantage of being able to call methods on model instances (instead of needing to import/"discover" free-floating functions), while helping reduce the potential for collisions going forward, and some of the current methods are named in ways that people have historically mentioned wanting to use as field names (e.g., `dict`, `json`, `schema`, etc.).

Related, it also allows us to add new methods following the same naming convention in the future knowing that they won't have conflicts. In principle you could keep the old names for old methods and use the new convention for new methods, but I think the conclusion is that given the inconsistency, this isn't the path we want to take.

As @samuelcolvin said, the plan is to retain the old behavior for some time with deprecation warnings, so it shouldn't cause problems to upgrade to v2.

**Reply by dmontagu** [reply]
> the new names if not ugly are verbose

I'll also add the point here that, at least in my experience, in the vast majority of my own business logic making use of pydantic models, I don't frequently call the pydantic-provided BaseModel methods; I much more frequently access field values or call my own custom-defined methods.

From that perspective, I think there is a benefit to freeing up the namespace and adding immediate visual clarity around which methods are "custom" and which are pydantic-provided.

**Comment by JoshYuJump** [comment]
> Why methods like .dict(), .json(), .json_schema() were renamed in V2 to .model_dump(), .model_dump_json(), .model_json_schema()?
> 
> This breaks backward compatibility a lot, and can be easily avoided. Also new names are very ugly, especially model_json_schema.

I Agree @dolfinus .
The only thing holding me back from using Pydantic v2 is these inelegant new methods.
